src/tracking.sml

```diff
@@ -0,0 +1,32 @@
+signature TRACKING =
+sig
+    val lineNum : int ref
+    val linePos : int list ref
+    val currentLineStartPos : int ref
+    val reset : unit -> unit
+    val incrementLine : int -> unit
+    val appendPosition : int -> unit
+    val getLinePos : int -> int
+end
```

Comment:
Seems like a lot of this is now duplicated in the error message struct too. What are the responsibilities of each of these structures now?